plotly/tools.py

```diff
@@ -2093,14 +2088,717 @@ def get_streamline_arrows(self):
+                                          'High', '') * len(decrease_x),
+                                    **kwargs)
+
+            return hidden_bar_decr, candle_bar_decr, candle_line_decr
+
+        layout = dict(barmode='stack',
+                      yaxis=dict(range=[(min(low) -
+                                        ((max(high) - min(low)) * .1)),
+                                        (max(high) +
+                                        ((max(high) - min(low)) * .1))]))
+
+        if direction is 'increasing':
+            (hidden_bar_incr,
+             candle_bar_incr,
+             candle_line_incr) = make_increasing_candle(increase_x,
+                                                        increase_open,
+                                                        increase_dif,
+                                                        stick_increase_y,
+                                                        stick_increase_x,
+                                                        **kwargs)
+
+            data = [hidden_bar_incr, candle_bar_incr, candle_line_incr]
+
+        elif direction is 'decreasing':
+            (hidden_bar_decr,
+             candle_bar_decr,
+             candle_line_decr) = make_decreasing_candle(decrease_x,
+                                                        decrease_close,
+                                                        decrease_dif,
+                                                        stick_decrease_y,
+                                                        stick_decrease_x,
+                                                        **kwargs)
+            data = [hidden_bar_decr, candle_bar_decr, candle_line_decr]
+
+        elif direction is 'both':
+            (hidden_bar_incr,
+             candle_bar_incr,
+             candle_line_incr) = make_increasing_candle(increase_x,
+                                                        increase_open,
+                                                        increase_dif,
+                                                        stick_increase_y,
+                                                        stick_increase_x,
+                                                        **kwargs)
+            (hidden_bar_decr,
+             candle_bar_decr,
+             candle_line_decr) = make_decreasing_candle(decrease_x,
+                                                        decrease_close,
+                                                        decrease_dif,
+                                                        stick_decrease_y,
+                                                        stick_decrease_x,
+                                                        **kwargs)
+            data = [hidden_bar_incr, candle_bar_incr, candle_line_incr,
+                    hidden_bar_decr, candle_bar_decr, candle_line_decr]
+
+        return dict(data=data, layout=layout)
+
+
+class _OHLC(dict):
```

Comment:
:cow2: Why does this need to inherit from `dict`? Wouldn't `object` suffice?

This is "private" though, so we can refactor whenever. nbd :)